### PR TITLE
Support conditionals

### PR DESCRIPTION
**Pull request by cicr99**

### CODE REVIEW

File: src/passes/expressionSplitter/conditionalFunctionaliser.ts [hunk 1]
```diff
@@ -165,3 +165,39 @@ export function addStatementsToCallFunction(
 export function getNodeVariables(node: Conditional) {
   return new Map([...collectUnboundVariables(node)].filter(([decl]) => !decl.stateVariable));
 }
+
+// There might be two different cases where conditionals return void:
+// - they are the expression of a return statement
+// - they are the expression of an expressionStatement
+// TODO - Check if there are any other cases where the conditional might be void
```
Comment (DomHenderson) [inline src/passes/expressionSplitter/conditionalFunctionaliser.ts:172]: They can also be nested in other void conditionals

File: src/passes/rejectUnsupportedFeatures.ts [hunk 2]
```diff
@@ -59,12 +58,6 @@ export class RejectUnsupportedFeatures extends ASTMapper {
   visitErrorDefinition(node: ErrorDefinition, _ast: AST): void {
     throw new WillNotSupportError('User defined Errors are not supported', node);
   }
-  visitConditional(node: Conditional, _ast: AST): void {
```
Comment (DomHenderson) [inline src/passes/rejectUnsupportedFeatures.ts:62]: This should still reject non-void tuple conditionals as they aren't supported (right?). It should also reject tuple assignments that are children of anything other than an expression statement

File: tests/behaviour/contracts/conditionals/conditional.sol [hunk 3]
```diff
@@ -0,0 +1,27 @@
+pragma solidity ^0.8.10;
+
+// SPDX-License-Identifier: MIT
+
+contract WARP {
+  uint256 public x;
+
+  function simpleScalar(
```
Comment (DomHenderson) [inline tests/behaviour/contracts/conditionals/conditional.sol:8]: I'd appreciate a test case that includes conditionals where both branches are literals. Ideally one for number literals as well as some for the various kinds of string literals.

**Review by DomHenderson — CHANGES_REQUESTED** [review]

**Comment by cicr99** [comment]
It was decided not to support conditionals yet due to many edge cases that can not be handled at the moment

**Pull request closed without merging**